Assignment6.Rmd

```diff
@@ -52,34 +59,106 @@ post(c.tree1, file = "tree1.ps", title = "MOOC") #This creates a pdf image of th
 
+```{r message = FALSE}
+#Get Student record table
+UMich_student.course <- read_csv("https://raw.githubusercontent.com/bkoester/PLA/master/student.course.csv")
+
+
+x <- filter(UMich_student.course, DIV == "O")
+```
+
+#### Codebook
+
+* ANONID - Anonymous ID
+* SUBJECT - Academic Subject
+* CATALOG_NBR - Course code? E.g. ACC 272
+* GRD_PTS_PER_UNIT - Grade Points Per Unit
+* GPAO - GPA in other classes
+* DIV - ACADEMIC DIVISION (E - Engineering, H - Humanities, O - Undergraduate Research, P - Business/Accounting/Finance, S - Science, SS - Social Science)
+* ANON_INSTR_ID - ANON INSTRUCTOR ID
+* TERM - Term number (see term.table.txt on the website)
+
+Will choose GPAO as outcome variable
+
 ```{r}
+#First Model
+#Use all available variables to predict GPAO
+c.tree_model_1 <- rpart(GPAO ~ CATALOG_NBR + SUBJECT + GRD_PTS_PER_UNIT + DIV + ANON_INSTR_ID + TERM,
+                        data = UMich_student.course)
+
+#Check the results from the classifcation tree using the printcp() command
+printcp(c.tree_model_1)
+
+#Plot your tree
+post(c.tree_model_1, file = "tree3.ps", title = "Model 1") #This creates a pdf image of the tree
+
+```
+
+```{r}
+#Second Model
+#Extract two variables to predict GPAO - Subject and term
+#If Grade inflation is a thing, later terms should have higher GPAO
+#If some subjects are easier to get As in, subject should predict GPAO (presuming that students taking a subject would be more likely to take other subjects 
+#in that area)
+c.tree_model_2 <- rpart(GPAO ~ SUBJECT + TERM,
+                        data = UMich_student.course)
+
+#Check the results from the classifcation tree using the printcp() command
+printcp(c.tree_model_2)
+
+#Plot your tree
+post(c.tree_model_2, file = "tree4.ps", title = "Model 2") #This creates a pdf image of the tree
 
 ```
 
+*Two models were used to predict Student's GPA in all other classes. The first model used all other variables. The second model used two variables - subject a student is enrolled in and the term.*
+
+*The first model generates 6 splits, with a 10-fold CV error of 0.67629 and standard error of 0.0032569. I decided not to prune the model given that this is the lowest error rate reported among all splits, and there was no substantial difference between the CV standard error in this level of the tree relative to other levels of the tree.*
+
+*The second model generates one split, with a 10-fold CV error of 0.9876 and standard error of 0.0035252. Since there was only one split with a negligible increase in standard error and a minimal decrease in 10-fold CV error, the tree at the level of 1 split was retained.*
+
+*Overall, the first model generates a better tree as it has a lower cross-validation error.*
```

Comment:
Very clear model comparison and selection. I like the part you explained why you would choose "subject" and "term" as the 2 predictors in your second model. In this case, the decision is relatively easy to make between the two models, you can further consider other cases when the error rate and standard error do not directly support one specific model and how would you make a decision in such a case.